miniapps/dpg/util/pml.hpp

```diff
@@ -0,0 +1,133 @@
+// Copyright (c) 2010-2022, Lawrence Livermore National Security, LLC. Produced
+// at the Lawrence Livermore National Laboratory. All Rights reserved. See files
+// LICENSE and NOTICE for details. LLNL-CODE-806117.
+//
+// This file is part of the MFEM library. For more information and source code
+// availability visit https://mfem.org.
+//
+// MFEM is free software; you can redistribute it and/or modify it under the
+// terms of the BSD-3 license. We welcome feedback and contributions, see file
+// CONTRIBUTING.md for details.
+
+#include "mfem.hpp"
+
+namespace mfem
+{
+
+/// Class for setting up a simple Cartesian PML region
+class CartesianPML
+{
+private:
+   Mesh *mesh;
+
+   /// Length of the PML Region in each direction
+   Array2D<double> length;
+
+   /// Computational Domain Boundary
+   Array2D<double> comp_dom_bdr;
+
+   /// Domain Boundary
+   Array2D<double> dom_bdr;
+
+   /** Integer Array identifying elements in the pml
+       0: in the pml, 1: not in the pml */
+   Array<int> elems;
+
+   /// Compute Domain and Computational Domain Boundaries
+   void SetBoundaries();
+
+public:
+   CartesianPML(Mesh *mesh_,Array2D<double> length_);
+
+   int dim;
+   double omega;
+   // Default values for Maxwell
+   double epsilon = 1.0;
+   double mu = 1.0;
+   /// Return Computational Domain Boundary
+   Array2D<double> GetCompDomainBdr() {return comp_dom_bdr;}
+
+   /// Return Domain Boundary
+   Array2D<double> GetDomainBdr() {return dom_bdr;}
```

Comment:
The layout of these arrays should be documented. Also there may be some ambiguity about their contents. I assume they contain coordinate values of where the boundaries are located but there are probably other possibilities.

One other point is that these return copies of the arrays whereas many other MFEM methods would return references. I imagine these are always very small so the copy is not an issue but I'm wondering if we might confuse some users. I expect you would never want the caller to alter the contents of these arrays. Perhaps they could return `const` references. Do you have use-cases in mind where copying into a locally owned array is preferable? I'm not opposed any return type that you feel is most logical. However, it may be wise to add a `@note` to the documentation explaining the choice.